File: src/02 - Random number/Random_number_qasm.py
from qiskit import *


def random_int(backend, size):
    circuit = QuantumCircuit.from_qasm_file("Random_number.qasm")

    job = execute(circuit, backend, shots=size, memory=True) 
    result = job.result()
    bin_num = ''.join(result.get_memory())
    return int(bin_num,2)

#QASM Simulator backend:
backend = Aer.get_backend('qasm_simulator')

#Real IBM Quantum Computer backend
#from qiskit.providers.ibmq import least_busy
#provider = IBMQ.load_account()
#backend = least_busy(provider.backends(filters=lambda x: x.configuration().n_qubits >= 3 and 
#                        not x.configuration().simulator and x.status().operational==True))


bits = 64
print("Here you have a random number: ", random_int(backend, bits))

File: src/02 - Random number/Random_number_qasm.jl
using PyCall

qiskit = pyimport("qiskit")

function random_int(backend, size)
	#Load circuit
	circuit = qiskit.QuantumCircuit.from_qasm_file("Random_number.qasm")
	job = qiskit.execute(circuit, backend, shots=size, memory=true)
	result = job.result()
	#Save result in string
	str = "0b"*join(result.get_memory())
	#Return base 10
	return parse(BigInt, str)
end

#QASM Simulator backend
backend = qiskit.Aer.get_backend("qasm_simulator")

bits = 64
print("Here you have a random number: ", random_int(backend, bits), "\n")
